plugins/jobs/web_client/views/JobListWidget.js

```diff
@@ -267,6 +314,31 @@ var JobListWidget = View.extend({
             };
             this.collection.on('g:changed', callback);
         });
+    },
+
+    _cancelJobs: function () {
+        Promise.all(
+            Object.keys(this.jobCheckedStates)
+                .filter(jobId => {
+                    var status = this.jobs.find(job => job.id === jobId).get('status');
+                    return [JobStatus.CANCELED, JobStatus.SUCCESS, JobStatus.ERROR].indexOf(status) === -1;
+                })
+                .map(jobId => Promise.resolve(restRequest({
+                    path: `job/${jobId}/cancel`,
+                    type: 'PUT',
+                    error: null
+                })))
+        ).then(results => {
+            if (!results.length) {
+                return;
+            }
+            events.trigger('g:alert', {
+                icon: 'ok',
+                text: `Cancel requests sent for ${results.length} ${results.length === 1 ? 'job' : 'jobs'}`,
```

Comment:
This might be a bit nicer: `${results.length} cancel ${results.length === 1 ? 'request' : 'requests'} sent`.